error_handling.md

```diff
@@ -0,0 +1,88 @@
+# Revamping error and none handling
+
+## Abstract
+This RFC proposes to change the current error handling mechanism, because the current `Option` type allows three values, `none`, `IError` and `T`. Handling three different kinds of values in every `or{}` block is hard and could lead to unwanted behavior.
+
+## Introduction
+It was several times discussed to change the current error handling / none handling mechanism because of the reason that `Option` allows three values.
+It would be better to break them down into two types and the developer specifies which one he actually wants to use.
+
+```v
+fn foo() ?string {...}
+
+foo() or {
+  // should I return a default value when none is returned or panic when an error is returned?
+}
+```
+The above example currently returns either a `string`, `none` or an `IError`, even if the function only returns `none` and `string` the end user might have to write code that also works with `IError` because the function could return it.
+
+> The new examples will use `Option` and `Result` to distinguish between `?` syntax. It's not the final syntax yet.
```

Comment:
`>` is treated by markdown as a citation mark

Reply:
I thought as a quote? What should I use as a hint?